data/src/test/java/org/apache/iceberg/io/TestTaskEqualityDeltaWriter.java

```diff
@@ -409,6 +421,55 @@ public void testUpsertDataWithFullRowSchema() throws IOException {
         .isEqualTo(ImmutableList.of(posRecord.copy("file_path", dataFile.path(), "pos", 0L)));
   }
 
+  @TestTemplate
+  public void testDeleteFileGranularity() throws IOException {
+    withGranularity(DeleteGranularity.FILE);
+  }
+
+  @TestTemplate
+  public void testDeletePartitionGranularity() throws IOException {
+    withGranularity(DeleteGranularity.PARTITION);
+  }
+
+  private void withGranularity(DeleteGranularity granularity) throws IOException {
+    List<Integer> eqDeleteFieldIds = Lists.newArrayList(idFieldId, dataFieldId);
+    Schema eqDeleteRowSchema = table.schema();
+
+    GenericTaskDeltaWriter deltaWriter =
+        createTaskWriter(eqDeleteFieldIds, eqDeleteRowSchema, granularity);
+
+    Map<Integer, Record> expected = Maps.newHashMapWithExpectedSize(2000);
+    // Create enough records, so we have multiple files
+    for (int i = 0; i < 2000; ++i) {
+      Record record = createRecord(i, "aaa" + i);
```

Comment:
maybe we create some position deletes via `update`. current monotonically increased index is unique record/key per iteration

Reply:
Added the 2 iteration of deletes, to prevent the monotonically increasing positions.
After based on your comment, I have fixed the test, and found the exact issue I was expecting first, and created the writer for handling it

Reply:
this still doesn't result in position deletes.

first round - `i % 5 == 0`
second round - `i * 10 + 1`

there is no overlap. both produces equality deletes?